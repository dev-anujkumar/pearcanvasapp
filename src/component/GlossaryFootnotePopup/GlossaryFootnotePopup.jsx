/**
* GlossaryFootnotePopup UI Component.
*/

import React, { Component } from 'react';
import Button from '../ElementButtons/ElementButton.jsx';
import '../../styles/GlossaryFootnotePopup/GlossaryFootnotePopup.css';
import { GLOSSARY } from './../../constants/Element_Constants';
import ReactEditor from "../tinyMceGlossaryFootnoteEditor"
import { checkforToolbarClick } from '../../js/utils'
import { hasReviewerRole } from '../../constants/utility.js'
import { setFormattingToolbar } from './GlossaryFootnote_Actions.js';
import AudioTinyMceGlossary from '../AudioTinyMceGlossary';
import AddAudioBook from '../AudioNarration/AddAudioBook.jsx';
import FigureTinyMceGlossary from '../ElementFigure/FigureTinyMceGlosaary.jsx'
import AddImageGlossary from '../ElementFigure/AddImageGlossary.jsx';

class GlossaryFootnotePopup extends Component {
    constructor() {
        super();
        this.state = {
            audioToggle:false,
            figureToggle:false
        }

    }

    handleAudioToggle =() =>{
        this.setState({
            audioToggle:!this.state.audioToggle
        })
    }
    closeAddAudioBook=()=>{
        this.setState({
            audioToggle:false
        })
    }

    handleFigureToggle =() =>{
        this.setState({
            figureToggle:!this.state.figureToggle
        })
    }
    closeFigurePopup=()=>{
        this.setState({
            figureToggle:false
        })
    }

    toolbarHandling = (e, action = "") => {
        let relatedTargets = (e && e.relatedTarget && e.relatedTarget.classList) ? e.relatedTarget.classList : [];
        if (e && checkforToolbarClick(relatedTargets)) {
            e.stopPropagation();
            return;
        }
        if (e && e.type == 'blur') {
            /** to disable both toolbars on glossary editor blur */
            setFormattingToolbar('disableTinymceToolbar')
            setFormattingToolbar('disableGlossaryFootnoteToolbar')
        }
        if (document.querySelector('div#toolbarGlossaryFootnote .tox-toolbar')) {
            if (action === "add") {
                setFormattingToolbar('disableGlossaryFootnoteToolbar')
            } else if (action === "remove") {
                setFormattingToolbar('disableTinymceToolbar')
                setFormattingToolbar('enableGlossaryFootnoteToolbar')
            }
        }
    }
    

    render() {
        const { glossaryFootnoteValue, closePopup, saveContent, permissions, glossaryFootNoteCurrentValue } = this.props;
        const glossaryFootnote = glossaryFootnoteValue.type;
        let id = glossaryFootnote === GLOSSARY ? 'glossary-1' : 'footnote-0';
        let accessToolbar = (permissions && permissions.includes('access_formatting_bar')) ? "" : " disableToolbar"

        let footnoteContentText = glossaryFootNoteCurrentValue.footnoteContentText;
        if (glossaryFootNoteCurrentValue.footnoteContentText && glossaryFootNoteCurrentValue.footnoteContentText.includes('imageAssetContent')) {
            let domparser = new DOMParser();
            let definitionDom = domparser.parseFromString(glossaryFootNoteCurrentValue.footnoteContentText, "text/html");
            let tempDiv = document.createElement('div');
            tempDiv.innerHTML = definitionDom.body?.childNodes[0]?.innerHTML;
            tinyMCE.$(tempDiv).find('img.imageAssetContent').remove();
            footnoteContentText = `<p>${tempDiv.innerHTML}</p>`;
        }

        return (
            <div ref={this.props.setWrapperRef} className="glossary-toolbar-wrapper">
                <div className="glossary-header">
                    <div className="glossary-title">
                        <Button type="close-container" onClick={saveContent} />
                        <div className="glossary-label">{glossaryFootnote}</div>
                        <div className="clr"></div>
                    </div>
                </div>
                <div id="toolbarGlossaryFootnote"></div>
                <div className="glossary-toolbar">
                {
                    glossaryFootnote === GLOSSARY &&<div className = {'audio-wrapper'+ accessToolbar} id='glossary-audio'><AudioTinyMceGlossary handleAudioToggle={this.handleAudioToggle} /></div>
                }

                {this.state.audioToggle && <AddAudioBook isGlossary={true} closeAddAudioBook={this.closeAddAudioBook} elementId={glossaryFootnoteValue.elementWorkId}/>}

                {
                    glossaryFootnote === GLOSSARY &&<div className = {'image-wrapper'+ accessToolbar} id='glossary-figure-image'><FigureTinyMceGlossary handleFigureToggle={this.handleFigureToggle} /></div>
                }
                {this.state.figureToggle && <AddImageGlossary  closeFigurePopup={this.closeFigurePopup} elementId={glossaryFootnoteValue.elementWorkId} />}
                </div>
                <div className="glossary-body">
                    <div id="glossary-toolbar"></div>
                    {
                        (glossaryFootnote === GLOSSARY) &&
                        <div className="glossary-word-header">
                            <div className="glossary-word-title">Term:</div>
                            <div className="glossary-word-name glossary-word-description" id='glossary-editor' onFocus={() => this.toolbarHandling(null, 'remove')} onBlur={(e) => this.toolbarHandling(e, 'add')}>
                                <ReactEditor permissions={this.props.permissions} glossaaryFootnotePopup={this.props.glossaaryFootnotePopup} glossaryFootNoteCurrentValue = {glossaryFootNoteCurrentValue.glossaryContentText} className='definition-editor place-holder' placeholder="Type Something" id='glossary-0' />
                            </div>
                        </div>
                    }
                    <div className="glossary-definition-header">
                        <div className="glossary-definition-label">{(glossaryFootnote === GLOSSARY) ? 'Definition:' : 'Note:'}</div>
                        <div className="glossary-editor glossary-definition-description" id="glossary-editor-attacher" onFocus={() => this.toolbarHandling(null, 'remove')} onBlur={(e) => this.toolbarHandling(e, 'add')}>
                            <ReactEditor permissions={this.props.permissions} glossaaryFootnotePopup={this.props.glossaaryFootnotePopup} glossaryFootNoteCurrentValue = {footnoteContentText} className='definition-editor place-holder' placeholder="Type Something" id={id} />
                        </div>
                    </div>
                    <div className="glossary-definition-buttons">
                        <span className="glossary-cancel-button" onClick={closePopup}>Cancel</span>
                        <span className="glossary-save-button" disabled={hasReviewerRole('elements_add_remove')} onClick={saveContent}>Save</span>
                    </div>
                </div>
            </div>
        )
    }

    componentWillUnmount() {
        
        for (let i = tinymce.editors.length - 1; i > -1; i--) {
            let ed_id = tinymce.editors[i].id;
            if (ed_id.includes('glossary') || ed_id.includes('footnote')) {
                /*
                    change wiris images to avoid converting to mathml
                */
               let tempContainerHtml = tinyMCE.$("#" + ed_id).html();          
               tempContainerHtml = tempContainerHtml.replace(/\sdata-mathml/g, ' data-temp-mathml').replace(/\"Wirisformula/g, '"temp_Wirisformula').replace(/\sWirisformula/g, ' temp_Wirisformula');
               document.getElementById( ed_id ).innerHTML = tempContainerHtml;
   
                tinymce.remove(`#${ed_id}`)
                tinymce.$('.wrs_modal_desktop').remove();
            }
        }
    }
}

export default GlossaryFootnotePopup;