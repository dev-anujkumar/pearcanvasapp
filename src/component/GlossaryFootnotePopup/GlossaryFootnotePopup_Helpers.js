//Constants
const PRIMARY = "primary";
const SECONDARY = "secondary";

//Focus Element
export const focusElement = (element, value) => {
    if (element && element.classList) {
        element.classList.add(value);
        element.tabIndex = "-1";
        element.focus();
    }
}

//Blur Element
export const blurElement = (element, value) => {
    if (element && element.classList) {
        element.classList.remove(value);
        element.tabIndex = "0";
        element.blur();
    }
}

//Get Primary Button
export const getPrimaryButton = () => {
    const primaryButton = document.getElementById("glossary-save-button") ? document.getElementById("glossary-save-button") : null;
    return primaryButton;
}

//Get Secondary Button
export const getSecondaryButton = () => {
    const secondaryButton = document.getElementById("glossary-cancel-button") ? document.getElementById("glossary-cancel-button") : null;
    return secondaryButton;
}

//Check if Primary Button is focused
export const isPrimaryButtonFocused = () => {
    const primaryButton = getPrimaryButton();
    return (primaryButton && primaryButton.classList && primaryButton.classList.contains(PRIMARY));
}

//Check if Secondary Button is focused
export const isSecondaryButtonFocused = () => {
    const secondaryButton = getSecondaryButton();
    return (secondaryButton && secondaryButton.classList && secondaryButton.classList.contains(SECONDARY));
}

//Focus Popup Buttons
export const focusPopupButtons = () => {
    if (!isPrimaryButtonFocused() && !isSecondaryButtonFocused()) {
        const secondaryButton = getSecondaryButton();
        setTimeout(() => {
            focusElement(secondaryButton, SECONDARY);
        }, 0);
    }
}

//Blur Popup Buttons
export const blurPopupButtons = () => {
    const primaryButton = getPrimaryButton();
    blurElement(primaryButton, PRIMARY);
    const secondaryButton = getSecondaryButton();
    blurElement(secondaryButton, SECONDARY);
}

//When Tab key is pressed
export const handleTabKeyPress = (e) => {
    const restrictedIdList = ["glossary-1", "footnote-0", "glossary-save-button", "glossary-cancel-button"];
    if (e && e.target && e.target.id && (restrictedIdList.indexOf(e.target.id) > -1)) {
        e.preventDefault();
    }
}

//When Enter key is pressed
export const handleEnterKeyPress = (primaryButton, secondaryButton) => {
    let element;
    if (isPrimaryButtonFocused()) {
        element = primaryButton;
    } else if (isSecondaryButtonFocused()) {
        element = secondaryButton;
    }
    if (element) {
        element.click();
    }
}

//When Esc key is pressed
export const handleEscKeyPress = (secondaryButton) => {
    if (secondaryButton) {
        secondaryButton.click();
    }
}

//Handle navigation on buttons
export const handleKeyDownHelper = (e) => {
    const primaryButton = getPrimaryButton();
    const secondaryButton = getSecondaryButton();
    if (e.keyCode === 9) {
        handleTabKeyPress(e);
    } else if (e.keyCode === 13) {
        handleEnterKeyPress(primaryButton, secondaryButton);
    } else if (e.keyCode === 27) {
        handleEscKeyPress(secondaryButton);
    } else if (e.keyCode === 37 && isPrimaryButtonFocused()) {
        blurElement(primaryButton, PRIMARY);
        focusElement(secondaryButton, SECONDARY);
    } else if (e.keyCode === 39 && isSecondaryButtonFocused()) {
        blurElement(secondaryButton, SECONDARY);
        focusElement(primaryButton, PRIMARY);
    }
}

//Check if audio or image figure button clicked
export const clickedOnAudioImageButtons = (e) => {
    const glossaryAudioButton = document.getElementById("glossary-audio") ? document.getElementById("glossary-audio") : null;
    const glossaryFigureImageButton = document.getElementById("glossary-figure-image") ? document.getElementById("glossary-figure-image") : null;
    return (e && e.target && ((glossaryAudioButton && glossaryAudioButton.contains(e.target)) || (glossaryFigureImageButton && glossaryFigureImageButton.contains(e.target))));
}

//Check if editor is clicked
export const clickedOnEditor = (e) => {
    const glossaryEditor = document.getElementById("glossary-editor") ? document.getElementById("glossary-editor") : null;
    const glossaryEditorAttacher = document.getElementById("glossary-editor-attacher") ? document.getElementById("glossary-editor-attacher") : null;
    return (e && e.target && ((glossaryEditor && glossaryEditor.contains(e.target)) || (glossaryEditorAttacher && glossaryEditorAttacher.contains(e.target))));
}

//Check if glossary asset popup is clicked
export const clickedOnOpenGlossaryAssetsPopup = (e) => {
    const openGlossaryAssetsPopup = document.getElementById("openFigureGlossary") ? document.getElementById("openFigureGlossary") : null;
    return (e && e.target && openGlossaryAssetsPopup && openGlossaryAssetsPopup.contains(e.target));
}

//Check if primary button is clicked
export const clickedOnPrimaryButton = (e) => {
    const primaryButton = getPrimaryButton();
    return (e && e.target && primaryButton && primaryButton.contains(e.target));
}

//Check if secondary button is clicked
export const clickedOnSecondaryButton = (e) => {
    const secondaryButton = getSecondaryButton();
    return (e && e.target && secondaryButton && secondaryButton.contains(e.target));
}

//Check if toolbar button is clicked
export const clickedOnToolbarButton = (e) => {
    const glossaryFootnoteToolbar = document.getElementById("toolbarGlossaryFootnote") ? document.getElementById("toolbarGlossaryFootnote") : null;
    return (e && e.target && glossaryFootnoteToolbar && glossaryFootnoteToolbar.contains(e.target));
}

//Handle click outside of editor
export const handleMouseDownHelper = (e) => {
    if (clickedOnAudioImageButtons(e) || clickedOnEditor(e) || clickedOnOpenGlossaryAssetsPopup(e) || clickedOnToolbarButton(e)) {
        blurPopupButtons();
    } else if (clickedOnPrimaryButton(e)) {
        const primaryButton = getPrimaryButton();
        const secondaryButton = getSecondaryButton();
        blurElement(secondaryButton, SECONDARY);
        focusElement(primaryButton, PRIMARY);
    } else if (clickedOnSecondaryButton(e)) {
        const primaryButton = getPrimaryButton();
        const secondaryButton = getSecondaryButton();
        blurElement(primaryButton, PRIMARY);
        focusElement(secondaryButton, SECONDARY);
    } else {
        focusPopupButtons();
    }
}