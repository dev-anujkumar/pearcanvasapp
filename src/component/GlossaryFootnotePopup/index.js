import GlossaryFootnoteMenu from './GlossaryFootnoteMenu';
export default GlossaryFootnoteMenu;