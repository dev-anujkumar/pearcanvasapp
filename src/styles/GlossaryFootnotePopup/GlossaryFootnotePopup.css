.glossary-toolbar-wrapper {
    background: #fff;
    position: absolute;
    top: 0;
    width: 100%;
    height:100%;
    z-index: 1000;
}

.glossary-toolbar-wrapper .glossary-title{
    margin: 0 15px;
}

.glossary-toolbar-wrapper .glossary-label{
    width: 75%;
    float: left;
    color:#fff;
    font-size: 18px;
    font-family: 'Open Sans', sans-serif;
    text-align: center;
    margin-top: -2px;
}

.glossary-toolbar-wrapper .glossary-body{
    margin: 0 12px;
}
.glossary-toolbar-wrapper #glossary-toolbar{
    background-color: #eee; 
    margin: -11px 0;
    border: 0;
    z-index: 10000;
}
.glossary-toolbar-wrapper .glossary-header{
    background-color: #000;
    padding: 14px 0 10px 0;
    z-index: 10000;
}
.glossary-word-header{
    background-color: #eee;
}
.glossary-toolbar-wrapper .glossary-word-title{
    padding: 10px 10px 5px;
    font-size: 16px;
    font-family: 'Open Sans', sans-serif;
}

.glossary-toolbar-wrapper .glossary-word-name{
    border-bottom: 2px solid #0080ff;
    color: #000;
    font-size: 14px;
    line-height: 28px;
    font-family: 'Open Sans', sans-serif;
    padding-top: 8px;
    padding-bottom: 8px;
}
.glossary-editor .glossary-definition-description{
    padding-bottom: 1px;
    padding-top: 2px;
}

.glossary-definition-header{
    background-color: #eee;
    margin-top: 20px;
    padding: 0 0 14px;
}
.glossary-toolbar-wrapper .glossary-definition-label{
    padding: 10px 10px 5px;
    font-size: 16px;
    font-family: 'Open Sans', sans-serif;
}
.glossary-toolbar-wrapper .glossary-definition-buttons{
    padding: 30px 10px 50px;
}
.glossary-toolbar-wrapper .glossary-cancel-button{
    padding: 5px 10px;
    background: #fff;
    color: #717272;
    margin-left: 10px;
    cursor: pointer;
    font-size: 16px;
    font-family: 'Open Sans', sans-serif;
    border: 1px solid #d9d9d9;
  }
  .glossary-toolbar-wrapper .glossary-save-button{
    padding: 5px 15px;
    background: #007ea3;
    color: #fff;
    margin-left: 10px;
    cursor: pointer;
    font-size: 16px;
    font-family: 'Open Sans', sans-serif;
}

.definition-editor {
    background-color: #eee;
    width: 90%;
    border: none;
    outline: none;
    margin: 0 11px 1px;
    font-size: 14px;
    font-family: 'Open Sans', sans-serif;
    line-height: 1.80em;
    letter-spacing: 0.01em;
    cursor:text;
}

/* ::placeholder{
    font-size: 14px;
    font-family: 'Open Sans', sans-serif;
    line-height: 1.80em;
    letter-spacing: 0.01em;
    margin: 0;
    outline: none;
} */
#toolbarGlossaryFootnote .tox-toolbar {
    background: transparent !important;
}
#toolbarGlossaryFootnote .tox-tbtn {
    border: 1px solid #ccc !important;
    margin: 2px 0px !important;
}
#toolbarGlossaryFootnote .tox .tox-tbtn {
    border-radius: 0 !important;
    width: 18%;
}

#toolbarGlossaryFootnote{
    margin: 11px 11px 11px !important;
    padding-bottom: 4%;
    border-bottom: solid 1px #e0e0e0;
}

#toolbarGlossaryFootnote .tox .tox-toolbar__group{
    padding: 1px 4px 2px 4px !important;
}
#toolbarGlossaryFootnote .fa {
    font: normal normal normal 14px/1px FontAwesome, sans-serif;
    font-family: FontAwesome, sans-serif;
}
#glossary-editor-attacher,#glossary-editor{
    min-height: 18px !important;
    max-height: 100px;
    overflow-x: auto;
}
#toolbarGlossaryFootnote>div{
    visibility: hidden;
    display: flex;
    border: 0;
}
.glossary-save-button[disabled] {
    opacity: 0.3;
    pointer-events: none;
}