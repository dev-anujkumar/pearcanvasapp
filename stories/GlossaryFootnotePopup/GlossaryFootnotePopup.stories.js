import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import  GlossaryFootnoteMenu  from '../../src/component/GlossaryFootnotePopup';
import { withInfo } from '@storybook/addon-info';

storiesOf('GlossaryFootnotePopup', module)
.addDecorator(withInfo)
.add('Footnote', () => <div style={{width:"240px", position:"relative", left:"100px"}}>
    <GlossaryFootnoteMenu glossaryFootnote="Footnote" closePopup={action('close')} saveContent={action('save')}/></div>)
.add('Glossary', () => <div style={{width:"240px", position:"relative", left:"100px"}}> 
<GlossaryFootnoteMenu glossaryFootnote="Glossary" closePopup={action('close')} saveContent={action('save')}/></div>)