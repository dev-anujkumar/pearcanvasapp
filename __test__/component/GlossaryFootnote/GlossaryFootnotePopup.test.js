import React from 'react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import GlossaryFootnotePopup from '../../../src/component/GlossaryFootnotePopup/GlossaryFootnotePopup';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('../../../src/js/utils',()=>{
return {
    checkforToolbarClick: ()=>{
        return false
    }
}
})
jest.mock('../../../src/component/GlossaryFootnotePopup/GlossaryFootnote_Actions', () => {
    return {
        setFormattingToolbar: () => { return; }
    }
})

jest.mock('../../../src/component/AudioTinyMceGlossary/AudioTinyMceGlossary', () => {
    return function () {
        return (<div>null</div>)
    }
})
jest.mock('../../../src/component/AudioNarration/AddAudioBook', () => {
    return function () {
        return (<div>null</div>)
    }
})

const initialState ={}
let store = mockStore(initialState);

describe('Testing GlossaryFootnote component with props', () => {
    let props={
        permissions:[],
        setWrapperRef: jest.fn(),
        showGlossaaryFootnote: jest.fn(),
        glossaryFootnoteValue:{"type":"","popUpStatus":false} ,
        closePopup:jest.fn(),
        saveContent:jest.fn(),
        glossaryFootNoteCurrentValue:{},
        audioGlossaryPopup:jest.fn()

    }
    let wrapper = mount(<Provider store={store}>< GlossaryFootnotePopup {...props} /></Provider>);
    let GlossaryFootnotePopupInstance = wrapper.find('GlossaryFootnotePopup').instance();


    it('render Glossary Footnote component -Footnote', () => {
        expect(wrapper).toHaveLength(1);
        expect(GlossaryFootnotePopupInstance).toBeDefined();
    })

    it('render Glossary Footnote component -Glossary', () => {
        let glossaryValue = {
            "type": "Glossary", "popUpStatus": false
        }
        let wrapper = mount(<Provider store={store}>< GlossaryFootnotePopup {...props} glossaryFootnoteValue={glossaryValue} /></Provider>)
        let GlossaryFootnotePopupInstance = wrapper.find('GlossaryFootnotePopup').instance();
        expect(wrapper).toHaveLength(1);
        expect(GlossaryFootnotePopupInstance).toBeDefined();
    })
    it('Test-toolbarHandling function -case2', () => {
        let wrapper = mount(<Provider store={store}>< GlossaryFootnotePopup {...props} /></Provider>, { attachTo: document.body })
        let GlossaryFootnotePopupInstance = wrapper.find('GlossaryFootnotePopup').instance();
        let event={
            relatedTarget:{
                classList:["tox-toolbar"]
            },
            stopPropagation: jest.fn()
        }
        let element = document.getElementById("toolbarGlossaryFootnote");
        element.innerHTML += "<div className= 'tox-toolbar'>B</div>";
        const spytoolbarHandling = jest.spyOn(GlossaryFootnotePopupInstance, 'toolbarHandling')
        GlossaryFootnotePopupInstance.toolbarHandling(event,"add")
        expect(spytoolbarHandling).toHaveBeenCalled() 
        spytoolbarHandling.mockClear()
    })
    
    it('Test-componentWillUnmount', () => {
        const spycomponentWillUnmount = jest.spyOn(GlossaryFootnotePopupInstance, 'componentWillUnmount')
        GlossaryFootnotePopupInstance.componentWillUnmount()
        expect(spycomponentWillUnmount).toHaveBeenCalled() 
        spycomponentWillUnmount.mockClear()
    })

    it('testcase for handleAudioToggle',()=>{
        GlossaryFootnotePopupInstance.handleAudioToggle();
        expect(GlossaryFootnotePopupInstance.state.audioToggle).toBe(true)
    })

    it('testcase for closeAddAudioBook',()=>{
        GlossaryFootnotePopupInstance.closeAddAudioBook();
        expect(GlossaryFootnotePopupInstance.state.audioToggle).toBe(false)
    })

})