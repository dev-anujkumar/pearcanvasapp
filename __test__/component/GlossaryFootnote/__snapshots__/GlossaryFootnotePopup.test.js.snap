// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GlossaryFootnote component with props render Footnote  1`] = `
<GlossaryFootnoteMenu
  glossaryFootnote="Footnote"
>
  <GlossaryFootnotePopup
    closePopup={[Function]}
    glossaryFootnote="Footnote"
    saveContent={[Function]}
  >
    <div
      className="glossary-toolbar-wrapper"
    >
      <div
        className="glossary-header"
      >
        <div
          className="glossary-title"
        >
          <ElementButton
            labelText="P"
            onClick={[Function]}
            type="close-container"
          >
            <span
              className="btn-element close-container"
              onClick={[Function]}
            >
              <img
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </span>
          </ElementButton>
          <div
            className="glossary-label"
          >
            Footnote
          </div>
          <div
            className="clr"
          />
        </div>
      </div>
      <div
        id="toolbarGlossaryFootnote"
      />
      <div
        className="glossary-body"
      >
        <div
          id="glossary-toolbar"
        />
        <div
          className="glossary-definition-header"
        >
          <div
            className="glossary-definition-label"
          >
            Note:
          </div>
          <div
            className="glossary-editor glossary-definition-description"
            id="glossary-editor-attacher"
          >
            <ReactEditor
              className="definition-editor"
              placeholder="Type Something"
            >
              <div
                className="definition-editor"
              >
                <Editor
                  cloudChannel="5"
                  init={
                    Object {
                      "fixed_toolbar_container": "#toolbarGlossaryFootnote",
                      "init_instance_callback": [Function],
                      "inline": true,
                      "menubar": false,
                      "selector": ".definition-editor",
                      "setup": [Function],
                      "toolbar": "bold | italic | underline | strikethrough | removeformat| superscript | subscript ",
                    }
                  }
                  onChange={[Function]}
                >
                  <div
                    id="tiny-react_53811291681568384695061"
                  />
                </Editor>
              </div>
            </ReactEditor>
          </div>
        </div>
        <div
          className="glossary-definition-buttons"
        >
          <span
            className="glossary-cancel-button"
            onClick={[Function]}
          >
            Cancel
          </span>
          <span
            className="glossary-save-button"
            onClick={[Function]}
          >
            Save
          </span>
        </div>
      </div>
    </div>
  </GlossaryFootnotePopup>
</GlossaryFootnoteMenu>
`;

exports[`Testing GlossaryFootnote component with props render Glossary  1`] = `
<GlossaryFootnoteMenu
  glossaryFootnote="Glossary"
>
  <GlossaryFootnotePopup
    closePopup={[Function]}
    glossaryFootnote="Glossary"
    saveContent={[Function]}
  >
    <div
      className="glossary-toolbar-wrapper"
    >
      <div
        className="glossary-header"
      >
        <div
          className="glossary-title"
        >
          <ElementButton
            labelText="P"
            onClick={[Function]}
            type="close-container"
          >
            <span
              className="btn-element close-container"
              onClick={[Function]}
            >
              <img
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </span>
          </ElementButton>
          <div
            className="glossary-label"
          >
            Glossary
          </div>
          <div
            className="clr"
          />
        </div>
      </div>
      <div
        id="toolbarGlossaryFootnote"
      />
      <div
        className="glossary-body"
      >
        <div
          id="glossary-toolbar"
        />
        <div
          className="glossary-word-header"
        >
          <div
            className="glossary-word-title"
          >
            Term:
          </div>
          <div
            className="glossary-word-name glossary-word-description"
            id="glossary-editor"
          >
            <ReactEditor
              className="definition-editor"
              placeholder="Type Something"
            >
              <div
                className="definition-editor"
              >
                <Editor
                  cloudChannel="5"
                  init={
                    Object {
                      "fixed_toolbar_container": "#toolbarGlossaryFootnote",
                      "init_instance_callback": [Function],
                      "inline": true,
                      "menubar": false,
                      "selector": ".definition-editor",
                      "setup": [Function],
                      "toolbar": "bold | italic | underline | strikethrough | removeformat| superscript | subscript ",
                    }
                  }
                  onChange={[Function]}
                >
                  <div
                    id="tiny-react_88003596761568384695046"
                  />
                </Editor>
              </div>
            </ReactEditor>
          </div>
        </div>
        <div
          className="glossary-definition-header"
        >
          <div
            className="glossary-definition-label"
          >
            Definition:
          </div>
          <div
            className="glossary-editor glossary-definition-description"
            id="glossary-editor-attacher"
          >
            <ReactEditor
              className="definition-editor"
              placeholder="Type Something"
            >
              <div
                className="definition-editor"
              >
                <Editor
                  cloudChannel="5"
                  init={
                    Object {
                      "fixed_toolbar_container": "#toolbarGlossaryFootnote",
                      "init_instance_callback": [Function],
                      "inline": true,
                      "menubar": false,
                      "selector": ".definition-editor",
                      "setup": [Function],
                      "toolbar": "bold | italic | underline | strikethrough | removeformat| superscript | subscript ",
                    }
                  }
                  onChange={[Function]}
                >
                  <div
                    id="tiny-react_85718437571568384695047"
                  />
                </Editor>
              </div>
            </ReactEditor>
          </div>
        </div>
        <div
          className="glossary-definition-buttons"
        >
          <span
            className="glossary-cancel-button"
            onClick={[Function]}
          >
            Cancel
          </span>
          <span
            className="glossary-save-button"
            onClick={[Function]}
          >
            Save
          </span>
        </div>
      </div>
    </div>
  </GlossaryFootnotePopup>
</GlossaryFootnoteMenu>
`;

exports[`Testing GlossaryFootnote component with props render Glossary Footnote component  1`] = `
<GlossaryFootnoteMenu
  glossaryFootnote="Glossary"
>
  <GlossaryFootnotePopup
    closePopup={[Function]}
    glossaryFootnote="Glossary"
    saveContent={[Function]}
  >
    <div
      className="glossary-toolbar-wrapper"
    >
      <div
        className="glossary-header"
      >
        <div
          className="glossary-title"
        >
          <ElementButton
            labelText="P"
            onClick={[Function]}
            type="close-container"
          >
            <span
              className="btn-element close-container"
              onClick={[Function]}
            >
              <img
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </span>
          </ElementButton>
          <div
            className="glossary-label"
          >
            Glossary
          </div>
          <div
            className="clr"
          />
        </div>
      </div>
      <div
        id="toolbarGlossaryFootnote"
      />
      <div
        className="glossary-body"
      >
        <div
          id="glossary-toolbar"
        />
        <div
          className="glossary-word-header"
        >
          <div
            className="glossary-word-title"
          >
            Term:
          </div>
          <div
            className="glossary-word-name glossary-word-description"
            id="glossary-editor"
          >
            <ReactEditor
              className="definition-editor"
              placeholder="Type Something"
            >
              <div
                className="definition-editor"
              >
                <Editor
                  cloudChannel="5"
                  init={
                    Object {
                      "fixed_toolbar_container": "#toolbarGlossaryFootnote",
                      "init_instance_callback": [Function],
                      "inline": true,
                      "menubar": false,
                      "selector": ".definition-editor",
                      "setup": [Function],
                      "toolbar": "bold | italic | underline | strikethrough | removeformat| superscript | subscript ",
                    }
                  }
                  onChange={[Function]}
                >
                  <div
                    id="tiny-react_79109996241568384695013"
                  />
                </Editor>
              </div>
            </ReactEditor>
          </div>
        </div>
        <div
          className="glossary-definition-header"
        >
          <div
            className="glossary-definition-label"
          >
            Definition:
          </div>
          <div
            className="glossary-editor glossary-definition-description"
            id="glossary-editor-attacher"
          >
            <ReactEditor
              className="definition-editor"
              placeholder="Type Something"
            >
              <div
                className="definition-editor"
              >
                <Editor
                  cloudChannel="5"
                  init={
                    Object {
                      "fixed_toolbar_container": "#toolbarGlossaryFootnote",
                      "init_instance_callback": [Function],
                      "inline": true,
                      "menubar": false,
                      "selector": ".definition-editor",
                      "setup": [Function],
                      "toolbar": "bold | italic | underline | strikethrough | removeformat| superscript | subscript ",
                    }
                  }
                  onChange={[Function]}
                >
                  <div
                    id="tiny-react_80872346151568384695014"
                  />
                </Editor>
              </div>
            </ReactEditor>
          </div>
        </div>
        <div
          className="glossary-definition-buttons"
        >
          <span
            className="glossary-cancel-button"
            onClick={[Function]}
          >
            Cancel
          </span>
          <span
            className="glossary-save-button"
            onClick={[Function]}
          >
            Save
          </span>
        </div>
      </div>
    </div>
  </GlossaryFootnotePopup>
</GlossaryFootnoteMenu>
`;
